misk/src/test/kotlin/misk/logging/DynamicLogLevelIntegrationTest.kt

```diff
@@ -0,0 +1,293 @@
+    assertThat(dynamicLevels).hasSize(3)
+    assertThat(dynamicLevels["com.amazonaws.services.dynamodbv2.AmazonDynamoDBLockClient"])
+      .isEqualTo(ch.qos.logback.classic.Level.TRACE)
+    assertThat(dynamicLevels["com.squareup.cash.dynamodb.lease.RealDynamoDbLease"])
+      .isEqualTo(ch.qos.logback.classic.Level.DEBUG)
+    assertThat(dynamicLevels["com.squareup.cash.roundtable"]).isEqualTo(ch.qos.logback.classic.Level.TRACE)
+  }
+
+  @Test
+  fun `service correctly parses and stores dynamic levels in filter`() {
+    // Configure with multiple loggers and different levels
+    featureFlags.override(
+      Feature("test-dynamic-logging"),
+      "com.amazonaws.services.dynamodbv2.AmazonDynamoDBLockClient=TRACE," +
+        "com.squareup.cash.dynamodb.lease=DEBUG," +
+        "com.squareup.cash.roundtable=TRACE",
+    )
+
+    service.updateLoggingLevels()
+
+    // Get the filter
+    val context = LoggerFactory.getILoggerFactory() as LoggerContext
+    val filter = context.turboFilterList.find { it.javaClass.simpleName == "LogbackLevelFilter" }
+
+    assertThat(filter).isNotNull
+
+    val dynamicLevels = getDynamicLevelsFromFilter(context)
+
+    // Verify the map contains exactly what we configured
+    assertThat(dynamicLevels).hasSize(3)
+    assertThat(dynamicLevels["com.amazonaws.services.dynamodbv2.AmazonDynamoDBLockClient"])
+      .isEqualTo(ch.qos.logback.classic.Level.TRACE)
+    assertThat(dynamicLevels["com.squareup.cash.dynamodb.lease"]).isEqualTo(ch.qos.logback.classic.Level.DEBUG)
+    assertThat(dynamicLevels["com.squareup.cash.roundtable"]).isEqualTo(ch.qos.logback.classic.Level.TRACE)
+  }
+
+  @Test
+  fun `service skips update when feature flag value unchanged`() {
+    val context = LoggerFactory.getILoggerFactory() as LoggerContext
+
+    // Set initial flag
+    featureFlags.override(Feature("test-dynamic-logging"), "com.test=DEBUG")
+    service.updateLoggingLevels()
+
+    val filter1 = context.turboFilterList.find { it.javaClass.simpleName == "LogbackLevelFilter" }
+
+    // Verify dynamicLevels contents
+    val dynamicLevels1 = getDynamicLevelsFromFilter(context)
+    assertThat(dynamicLevels1).hasSize(1)
+    assertThat(dynamicLevels1["com.test"]).isEqualTo(ch.qos.logback.classic.Level.DEBUG)
+
+    // Set same flag again
+    featureFlags.override(Feature("test-dynamic-logging"), "com.test=DEBUG")
+    service.updateLoggingLevels()
+
+    val filter2 = context.turboFilterList.find { it.javaClass.simpleName == "LogbackLevelFilter" }
+
+    // Should be the same filter instance (not replaced)
+    assertThat(filter1).isSameAs(filter2)
+
+    // Verify dynamicLevels contents remain the same
+    val dynamicLevels2 = getDynamicLevelsFromFilter(context)
+    assertThat(dynamicLevels2).hasSize(1)
+    assertThat(dynamicLevels2["com.test"]).isEqualTo(ch.qos.logback.classic.Level.DEBUG)
+  }
+
+  @Test
+  fun `service processes feature flag with whitespace`() {
```

Comment:
This test seems redundant to `DynamicLogParser.handles whitespace around entries`

Reply:
Good catch. Will remove the test.